src/server/scripts/Pet/pet_hunter.cpp

```diff
@@ -95,20 +95,20 @@ class npc_pet_hunter_snake_trap : public CreatureScript
                     Unit* summoner = me->ToTempSummon()->GetSummonerUnit();
 
                     std::vector<Unit*> targets;
-                    for (std::pair<ObjectGuid const, PvPCombatReference*> const& pair : summoner->GetCombatManager().GetPvPCombatRefs())
+
+                    auto addTargetIfValid = [&](CombatReference* ref)
```

Comment:
capture `targets` explicitly

Reply:
You mean something like that? 
```cpp
auto addTargetIfValid = [this, targets, summoner](CombatReference* ref) mutable
```

Reply:
yes, ensure you capture `targets` by reference